### PR TITLE
soc: arm: stm32u5 device with DCACHE for writing to NOR flash

### PR DESCRIPTION
**Pull request by FRASTM**
Enable the DCACHE1 in INCR burst mode to allow writing to the external NOR octoFlash when in MemoryMapped mode.

This PR will help XiP from external NOR on the stm32u5 targets.

### CODE REVIEW

File: soc/arm/st_stm32/stm32u5/soc.c
Comment (erwango) [inline soc/arm/st_stm32/stm32u5/soc.c:41]: What are the impact on existing code when enabling DCACHE systematically? For instance use of DMA ?

Also, rather than enabling it directly here, I think we should implement a cache driver (see `drivers/cache`), in order to allow wusage of system API `sys_cache_data_foo` from drivers.

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32u5/soc.c
Comment (FRASTM) [reply soc/arm/st_stm32/stm32u5/soc.c:41]: with

> What are the impact on existing code when enabling DCACHE systematically? For instance use of DMA ? At least I think it should be enabled conditionaly
> 
> Also, rather than enabling it directly here, I think we should implement a cache driver (see `drivers/cache`), in order to allow wusage of system API `sys_cache_data_foo` from drivers.

With 
```
	sys_cache_instr_enable();
	sys_cache_data_enable();
```

both testcase tests/drivers/dma/ : PASSED

**Comment by aescolar** [comment]
Bulk changing the milestone from everything which did not make it to the v3.7.0 freeze deadline
and which is not labeled as a bug or documentation (or does not appear clearly as only containing
a bug fix or a documentation change).
If you disagree about this change please add the milestone again.

Please note that until rc2 only PRs containing bug fixes, docs and tests can be merged.
After that and until release only bug fixes and docs changes.

Exceptions require approval by the release team and a vote by the TSC.
https://docs.zephyrproject.org/latest/project/release_process.html#stabilization-phase

**Pull request closed without merging**
